api/src/main/java/io/onetable/spi/sync/TableFormatSync.java

```diff
@@ -69,35 +91,57 @@ public SyncResult syncSnapshot(OneSnapshot snapshot) {
    * @param changes the changes from the source table format that need to be applied
    * @return the results of trying to sync each change
    */
-  public List<SyncResult> syncChanges(IncrementalTableChanges changes) {
-    List<SyncResult> results = new ArrayList<>();
-    for (TableChange change : changes.getTableChanges()) {
-      Instant startTime = Instant.now();
-      try {
-        results.add(
-            getSyncResult(
-                SyncMode.INCREMENTAL,
-                change.getTableAsOfChange(),
-                client -> client.syncFilesForDiff(change.getFilesDiff()),
-                startTime,
-                changes.getPendingCommits()));
-      } catch (Exception e) {
-        // Fallback to a sync where table changes are from changes.getInstant() to latest, write a
-        // test case for this.
-        // (OR) Progress with empty col stats to mirror the timeline.
-        log.error("Failed to sync table changes", e);
-        results.add(buildResultForError(SyncMode.INCREMENTAL, startTime, e));
-        break;
+  public Map<TableFormat, List<SyncResult>> syncChanges(
+      Map<TargetClient, Optional<OneTableMetadata>> filteredSyncMetadataByClient,
+      IncrementalTableChanges changes) {
+    Map<TableFormat, List<SyncResult>> results = new HashMap<>();
+    Set<TargetClient> clientsWithFailures = new HashSet<>();
+    while (changes.getTableChanges().hasNext()) {
```

Comment:
target specific syncs are leaking into this. We should probably have a method in target client or something else.

Reply:
I don't understand what you're saying here

Reply:
I'm saying whether a table change should be synced or not based on some filters should probably reside in TargetClient so this part doesn't care about specificity of target clients.

Reply:
Ok, can we tackle that in some other PR then? This is merely moving existing code from OneTableClient to the TableFormatSync.